engine/runtime/src/main/java/org/enso/interpreter/node/callable/function/BlockNode.java

```diff
@@ -55,8 +57,15 @@ public static BlockNode buildSilent(ExpressionNode[] expressions, ExpressionNode
   @Override
   @ExplodeLoop
   public Object executeGeneric(VirtualFrame frame) {
+    var ctx = EnsoContext.get(this);
+    var nothing = ctx.getBuiltins().nothing();
     for (ExpressionNode statement : statements) {
-      statement.executeGeneric(frame);
+      var result = statement.executeGeneric(frame);
+      if (result != nothing) {
+        if (result instanceof DataflowError err) {
+          return err;
```

Comment:
@radeusgd suggests to capture the whole stacktrace here (in addition to existing `DataflowError` location) as this is an exceptional state anyway.

Reply:
It would be easier to trace down where the error is (rather unexpectedly) rethrown.

This is based on my attempts at fixing the test errors this change has uncovered.

Reply:
Why not just convert it to `PanicError`  then? Just call something like `EnsoContext.raiseAssertionError`?

Reply:
That makes it fully exceptional indeed. If we want set a coding standard that the dataflow errors should never go unhandled, that would work OK.

But actually writing code that relies on this new auto-propagation of errors could allow us to simplify A LOT. For example,
```
a = f.copy_to g
a.if_not_error g
```
could become just
```
f.copy_to g
g
```
which is much simpler (the example above is simple overall but we have big chunks of code that are tied together with `if_not_error` that could get **vastly** simplified with the new propagation scheme).

Well, I guess you made me prove that actually we want to rely on this new propagation schema as _not that exceptional_, less than I initially thought.

Reply:
> want to rely on this new propagation schema as _not that exceptional_, less than I initially thought.

OK, in such case there is going to be no change in stacktrace manipulation in this PR.